### PR TITLE
crypto, keypair: move keys to subpackages

### PR DESCRIPTION
**Pull request by noot**
## Changes
- moved ed25510 and sr25519 key implementations to their own subpackages
- fixed keystore and crypto packages, are more decoupled now

## Tests:
```
go test ./crypto/... ./keystore/
```

### CODE REVIEW

File: cmd/gossamer/account.go [hunk 1]
```diff
@@ -164,13 +166,13 @@ func generateKeypair(keytype, datadir string, password []byte) (string, error) {
 	var err error
 	if keytype == crypto.Sr25519Type {
 		// generate sr25519 keys
-		kp, err = crypto.GenerateSr25519Keypair()
+		kp, err = sr25519.GenerateKeypair()
```
Comment (ansermino) [inline cmd/gossamer/account.go]: We should probably update `generateKeypair` to take ` crypto.KeyType`

File: runtime/wasmer_test.go [hunk 2]
```diff
@@ -31,9 +31,11 @@ import (
 
 	"github.com/ChainSafe/gossamer/common"
 	"github.com/ChainSafe/gossamer/crypto"
+	"github.com/ChainSafe/gossamer/crypto/ed25519"
+	"github.com/ChainSafe/gossamer/crypto/sr25519"
 	"github.com/ChainSafe/gossamer/keystore"
 	"github.com/ChainSafe/gossamer/trie"
-	"golang.org/x/crypto/ed25519"
+	//"golang.org/x/crypto/ed25519"
```
Comment (ansermino) [inline runtime/wasmer_test.go]: remove

File: cmd/gossamer/account.go (hunk 1 above)
Comment (noot) [reply cmd/gossamer/account.go]: why? it's now part of the subpackage so it's specific to each key implementation

File: runtime/wasmer_test.go (hunk 2 above)
Comment (noot) [reply runtime/wasmer_test.go]: done

**Pull request merged**